File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        minPrice = prices[0]
        maxProfit = 0
        n= len(prices)
        for i in range(0, n):
     
            minPrice = min(prices[i], minPrice)
            maxProfit = max(maxProfit, prices[i] - minPrice)

        return maxProfit
        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length;
        int ans=0;
        int start =0;
        int end =1;
        while(start<n && end<n){
            if(prices[start]>prices[end]){
                start = end;
                end++;
            }else{
                ans = Math.max(prices[end]-prices[start],ans);
                end++;
            }
        }
        return ans;

    }
}